### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by warniaha**
Jenkins job: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/rgwdaos-test/

Docker host: [ssc-vm-g3-rhev4-3251.colo.seagate.com](http://ssc-vm-g3-rhev4-3251.colo.seagate.com/)

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/daos/docker/ceph/ccache.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+max_size = 25G 
+sloppiness = time_macros
+run_second_cpp = true
```
Comment (zalsader) [inline src/daos/docker/ceph/ccache.conf:3]: No newline at end of file, a bunch of other files have this too.

File: src/daos/docker/daos/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RUN sudo dnf install openssl git ccache jq net-tools iproute -y
+RUN sudo dnf update -y
+
+# copy the daos yml files for a local run
+COPY docker_daos_server.yml /opt/daos/etc/daos_server.yml
+COPY docker_daos_agent.yml /opt/daos/etc/daos_agent.yml
+COPY docker_daos_control.yml /opt/daos/etc/daos_control.yml
+ENV DAOS_PATH=/opt/daos
+ENV SOURCE_DATE_EPOCH=946684800
+RUN sudo bash -c 'echo "vm.nr_hugepages = 513" >> /etc/sysctl.conf'
```
Comment (zalsader) [inline src/daos/docker/daos/Dockerfile:22]: 513?

File: src/daos/docker/test.sh [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
+}
+
+function start_daos_cortx_s3tests()
+{
+    # check for anything running as this may be a re-run
+    PROCESSES=$(docker exec $CONTAINER_NAME  ps -e)
+    if [[ ! $PROCESSES =~ daos_server ]]; then
+        docker exec -u 0 $CONTAINER_NAME /opt/daos/bin/daos_server start &
+        if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
```
Comment (zalsader) [inline src/daos/docker/test.sh:33]: There is a lot of repeated code here, could be abstracted

File: src/daos/docker/update_confluence.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+   # ],
+    "space": {
+        "id": 458755,
+        "key": "PUB",
+    },
+   "body": {
+      #  "view": {
+      #    "value": "<string>",
+      #    "representation": "view"
+      #  },
```
Comment (zalsader) [inline src/daos/docker/update_confluence.py:42]: What's up with the commented out code here?

**Review by zalsader — APPROVED** [review]
All of these are great, thank you for working on this. I included a few notes about cleaning up the code. There is a chance we could move all these files to a new repository later down the line - is there anything that could prevent this?

File: src/daos/docker/daos/Dockerfile (hunk 2 above)
Comment (warniaha) [reply src/daos/docker/daos/Dockerfile:22]: something else is using 1 hugepage, causes DAOS to fail

File: src/daos/docker/test.sh (hunk 3 above)
Comment (warniaha) [reply src/daos/docker/test.sh:33]: just the error handling.  How would you suggest this be modified?

File: src/daos/docker/update_confluence.py (hunk 4 above)
Comment (warniaha) [reply src/daos/docker/update_confluence.py:42]: my bad, I guess I forgot to remove them

File: src/daos/docker/build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+cd daos
+docker build . -t "daos-single-host"
+if [[ ! $? == 0 ]]; then exit 1; fi
+cd ../ceph
+docker build . -t "dgw-single-host"
+if [[ ! $? == 0 ]]; then exit 1; fi
+cd ../s3-tests
+docker build . -t "dgw-s3-tests"
+if [[ ! $? == 0 ]]; then exit 1; fi
+sh test.sh
```
Comment (zalsader) [inline src/daos/docker/build.sh:16]: Could you please make sure each file has a newline at the end? You can also use vscode format

**Pull request merged**